Makefile.toml

```diff
@@ -166,19 +167,65 @@ dependencies = ["setup-build"]
 script_runner = "bash"
 script = [
 '''
-set -o pipefail
-if ! docker image inspect ${BUILDSYS_SDK_IMAGE} >/dev/null 2>&1 ; then
-  # Let curl resolve the certificates instead of the tasks resolved bundle.
-  unset SSL_CERT_FILE SSL_CERT_DIR
-  if ! curl --silent --fail --show-error https://${BUILDSYS_SDK_SITE}/${BUILDSYS_SDK_IMAGE}.tar.gz \
-       | gunzip | docker load ; then
-    echo "failed to load '${BUILDSYS_SDK_IMAGE}'" >&2
+if ! docker image inspect "${BUILDSYS_SDK_IMAGE}" >/dev/null 2>&1 ; then
+  if ! docker pull "${BUILDSYS_SDK_IMAGE}" ; then
+    echo "failed to pull '${BUILDSYS_SDK_IMAGE}'" >&2
     exit 1
   fi
 fi
 '''
 ]
 
+[tasks.fetch-toolchain]
+dependencies = ["setup-build"]
+script_runner = "bash"
+script = [
+'''
+# This is complicated because we want the image with the target's native toolchain,
+# which may not match the host architecture, so we can't just pull it directly.
```

Comment:
Can you pull it directly with `--platform`?  For example:

```
$ docker pull --platform arm64 public.ecr.aws/c2b5m1s5/thar-be-a-beta-toolchain-aarch64:v0.20.0
v0.20.0: Pulling from c2b5m1s5/thar-be-a-beta-toolchain-aarch64
e866ce65966e: Pull complete 
Digest: sha256:81ce1722abcb175b7cc4159f35ace4cddaa2de6e21b35b7590a29ecb34d96371
Status: Downloaded newer image for public.ecr.aws/c2b5m1s5/thar-be-a-beta-toolchain-aarch64:v0.20.0
public.ecr.aws/c2b5m1s5/thar-be-a-beta-toolchain-aarch64:v0.20.0
$ docker run -it --rm public.ecr.aws/c2b5m1s5/thar-be-a-beta-toolchain-aarch64:v0.20.0 sh
WARNING: The requested image's platform (linux/arm64) does not match the detected host platform (linux/amd64) and no specific platform was requested
docker: Error response from daemon: OCI runtime create failed: container_linux.go:367: starting container process caused: exec: "sh": executable file not found in $PATH: unknown.
```

That pulled the aarch64 image and (as expected) failed to run on my x86_64 host.

Reply:
TIL about `--platform` 😄 - I will give it a try!